subscription_manager/controller.py

```diff
@@ -64,34 +69,35 @@ def delete_subscription(self, subscription_id: int) -> int:
         """
         pass
 
-    def get_subscription_by_id(self, subscription_id: int) -> dict:
+    def get_subscription_by_name(self, subscription_name: str) -> Subscription:
         """
-        Return subscription by identifier
+        Return subscription by name
         Returns:
-            dict: found subscription in dict representation
+            <class Subscription>: found subscription like Subscription object
         Raises:
             SubsNotFoundException: when subscription with this id doesn't exist in database
         """
-        pass
+        if type(subscription_name) is not str:
+            raise WrongTypeException(
+                f"Found wrong field 'subscription_name' type, expected:<str>, received:{type(subscription_name), subscription_name}"
+            )
+        if len(subscription_name) < 1:
```

Comment:
if not (subscription_name and isinstance(subscription_name, str)):
    raise exception

Reply:
There are different type of exception in case of wrong name type and name value, an empty string is a wrong value, None and integer are wrong types.